projects/ngx-diagrams/src/lib/models/diagram.model.ts

```diff
@@ -129,9 +129,24 @@ export class DiagramModel extends BaseEntity {
 	}
 
 	reset() {
-		Object.values(this.getNodes()).forEach(node => {
-			this.deleteNode(node);
-		});
+		const ports = Array.from(this.getAllPorts().values());
+		const links = Object.values(this.getLinks());
+		const nodes = Object.values(this.getNodes());
+
+		for (const node of nodes) {
+			node.destroy();
+		}
+
+		for (const port of ports) {
```

Comment:
I would have add this inside the node destruction method (which doesn't exist at the moment), since a port or ports are direct children of a node.

Reply:
Agreed